src/util/builders/url/api.director.ts

```diff
@@ -55,4 +55,11 @@ export default class ApiURLDirector {
       .setQueryParams({ action: 'parse', format: 'json', prop: 'modules|jsconfigvars|headhtml', formatversion: '2', page: articleId })
       .build()
   }
+
+  buildGadgetsUrl() {
```

Comment:
The API queries `buildGadgetsUrl`, `buildSiteInfoURL` and `buildNamespacesURL` are all only fetched once in `mwoffliner.lib.ts`, those three requests could be combined into a single one.

Reply:
I've pushed the matter to a distinct issue: https://github.com/openzim/mwoffliner/issues/2385

Both for clarity of review and because I'm not yet sure about the exact impact of this move. And anyway, it is not really an urgent change from my perspective, more a kind of optimization